libr/core/cmd.c

```diff
@@ -6722,14 +6756,16 @@ static RCmdStatus core_cmd_tsr2cmd(RCore *core, const char *cstr, bool split_lin
 	R_LOG_DEBUG("s-expr %s\n", ts_str);
 	free (ts_str);
 
+	state.ac = __core_arroba_context_new (state.core);
```

Comment:
The "proper" way to fix `@v` in newshell would be to add it to https://github.com/radareorg/radare2/blob/master/shlr/radare2-shell-parser/grammar.js#L294 . For example, add a:
```js
tmp_value_command: $ => prec.right(1, seq($._simple_command, '@v:', $.arg)),
```
and add `tmp_value_command` in https://github.com/radareorg/radare2/blob/master/shlr/radare2-shell-parser/grammar.js#L124 .

Then follow the instructions at https://github.com/radareorg/radare2/blob/master/shlr/radare2-shell-parser/README.md to re-compile the grammar.

At that point you can add `tmp_value_command` to https://github.com/radareorg/radare2/blob/master/libr/core/r2-shell-parser-cmds.inc and then add in this file a `DEFINE_HANDLE_TS_FCN_AND_SYMBOL(tmp_value_command) {` similar to the others.

Reply:
Thanks @ret2libc im still alien to the treesitter code. i'll try it out and update the PR. still the current foreach parsing code from cmd.c needs to refacted out to be reused from newshell. the @@ are also missing